src/source/Signaling/Signaling.c

```diff
@@ -450,10 +451,20 @@ STATUS signalingFetchSync(PSignalingClient pSignalingClient)
 
     // move to the fromGetToken() so we can move to the necessary step
     setStateMachineCurrentState(pSignalingClient->pStateMachine, SIGNALING_STATE_GET_TOKEN);
+
+    //if we're not failing from a bad token, set the result to OK to that fromGetToken will move
+    //to getEndpoint, describe, or create. If it is bad, keep reiterating on token.
+    result = ATOMIC_LOAD(&pSignalingClient->result);
+    if (result != SERVICE_CALL_NOT_AUTHORIZED) {
+        ATOMIC_STORE(&pSignalingClient->result, (SIZE_T) SERVICE_CALL_RESULT_OK);
+    }
```

Comment:
From a design perspective I really don't like modifying the result here it seems very hacky.  Why is this necessary?  What are the alternatives?